### ISSUE TITLE
1.9.0 has different error than 1.8.0 for revoked client cert

### ISSUE THREAD

**Issue by pgr0ss** [grpc/grpc-java#4016]
Please answer these questions before submitting your issue.

### What version of gRPC are you using?

1.9.0

### What did you expect to see?

We have a test which asserts that we get an error when trying to use a revoked client cert. On 1.8.0, we get a `StatusRuntimeException` on the call, and this exception has a cause with a nice message. Our test looks like this:

```java
  @Test
  public void failsWithRevokedClient() throws SSLException {
    try {
      callWithClientCert("revoked.client.crt", "revoked.client.key.pkcs8");
      fail();
    } catch (StatusRuntimeException e) {
      assertEquals(Status.Code.UNAVAILABLE, e.getStatus().getCode());
      assertThat(e.getCause(), instanceOf(SSLHandshakeException.class));
      assertThat(e.getCause().getMessage(), new StringContains("CERTIFICATE_REVOKED"));
    }
  }
```

This works on 1.8.0, but now on 1.9.0, the `e.getCause()` returns `null`. So the request still fails, but we can't assert that it's failing for the right reasons.

Did something change around revoked certs or error handling in 1.9.0?

I didn't see any tests in grpc-java that seem to test an explicitly revoked client certificate. I did see this message for untrusted certs:

https://github.com/grpc/grpc-java/blob/ff0ad5fac3cf944a5ce5fb3ccef7df4a2d0bdf14/interop-testing/src/test/java/io/grpc/testing/integration/TlsTest.java#L187-L188

**Comment by ejona86** [comment]
Can you paste the errors you get in each version?

**Comment by ejona86** [comment]
Interesting. This may have been fixed by #3997. That'd be interesting because if so, this is likely a regression in Netty.

@pgr0ss, is it easy for you to test against master? I just created a fresh 1.10.0-SNAPSHOT in https://oss.sonatype.org/content/repositories/snapshots/ if that helps.

**Comment by pgr0ss** [comment]
@ejona86 I was able to test against master and it does seem like the cause is back:

```
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception                                                         
javax.net.ssl.SSLHandshakeException: error:10000414:SSL routines:OPENSSL_internal:SSLV3_ALERT_CERTIFICATE_REVOKED 
```

The `StatusRuntimeException` has a new piece of the message (`: io exception`) but that seems totally fine.

Any chance this fix can be backported to a 1.9.1? Thanks!

**Comment by ejona86** [comment]
This should be reasonably feasible to backport. There's a proxy fix that we'll probably be making a release for anyway. The fix didn't add a test though, so I do want to add one.

**Comment by carl-mastrangelo** [comment]
@pgr0ss 1.10.0 is going to be cut this week, and deployed in about 2 weeks.   Would you be willing to wait for the next release?

**Comment by pgr0ss** [comment]
@carl-mastrangelo Yeah, that would work for us.

**Comment by ejona86** [comment]
I'm backporting the fix to 1.9.x in #4033 and it seems likely we'll make a patch release for a proxy problem (after I do some manual testing). Closing since this is fixed on master and the 1.10.x branch.

**Comment by ejona86** [comment]
1.9.1 is released with this fixed and available on Maven Central.
